akka-persistence-testkit/src/main/scala/akka/persistence/testkit/PersistenceTestKitPlugin.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2018-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.testkit
+
+import akka.annotation.InternalApi
+import akka.persistence._
+import akka.persistence.journal.AsyncWriteJournal
+import akka.persistence.snapshot.SnapshotStore
+import akka.persistence.testkit.internal.{ InMemStorageExtension, SnapshotStorageEmulatorExtension }
+import com.typesafe.config.{ Config, ConfigFactory }
+
+import scala.collection.immutable
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+ * INTERNAL API
+ *
+ * Persistence testkit plugin for events.
+ */
+@InternalApi
+class PersistenceTestKitPlugin extends AsyncWriteJournal {
+
+  private final val storage = InMemStorageExtension(context.system)
+
+  override def asyncWriteMessages(messages: immutable.Seq[AtomicWrite]): Future[immutable.Seq[Try[Unit]]] =
+    Future.fromTry(Try(messages.map(aw => storage.tryAdd(aw.payload))))
```

Comment:
somewhere the `Tagged` should be unwrapped, so that asyncReplayMessages doesn't return the `Tagged`, as in the issue fixed in https://github.com/akka/akka/pull/28553

Reply:
This is linked to #28602. I for now, until tck, and possibly some improvements to journal mechanics are made, I am planning just to ignore tags. Is that ok?

Reply:
Would be nice if you unwrapped the `Tagged` before `tryAdd` here. In that way they are ignored, but it still works to use them for writing/replay.